dj/sql/parsing/ast.py

```diff
@@ -220,25 +233,98 @@ class Expression(Node):
     """an expression type simply for type checking"""
 
 
-@dataclass(eq=False)  # type: ignore
-class Named(Expression):
-    """An Expression that has a name"""
+@dataclass(eq=False)
+class Name(Node):
+    """
+    the string name specified in sql with quote style
+    """
 
     name: str
     quote_style: str = ""
 
-    @property
-    def quoted_name(self) -> str:
-        """
-        get the name of the Named Node including the quotes if any
-        """
+    def __str__(self) -> str:
         return (
             f"{self.quote_style}{self.name}{self.quote_style}"  # pylint: disable=C0301
         )
 
+    def to_column(self) -> "Column":
+        """
+        transform the name into a column
+        """
+        return cast(Column, Column(self).add_self_as_parent())
+
+    def to_table(self) -> "Table":
+        """
+        transform the name into a Table
+        """
+        return cast(Table, Table(self).add_self_as_parent())
+
+    def to_namespace(self) -> "Namespace":
+        """
+        transforms a single Name to a single item Identifier
+        """
+        return cast(Namespace, Namespace([self]).add_self_as_parent())
+
+    def __hash__(self) -> int:
+        return hash(self.name + self.quote_style)
+
+
+@dataclass(eq=False)
+class Namespace(Node):
+    """
+    Represents a sequence of names prececeding some Table or Column
+    """
+
+    names: List[Name]
+
+    def to_column(self) -> "Column":
+        """
+        transform the namespace into a column
+            whose name is the last name in the namespace
```

Comment:
Nit: the convention for docstrings (https://peps.python.org/pep-0257/#multi-line-docstrings) is to have the summary description in a single line. There are tools that will index the summary, and if it's spread in two lines they'll show only the first half.